OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -141,6 +141,10 @@ public static void NotifyBlocker(this Actor self, IEnumerable<CPos> positions)
 					return false;
 			}
 
+			// Occupied by any actor, including ally/friendly units
+			if (self.World.ActorMap.GetActorsAt(pos).Any())
```

Comment:
I thought it was a good idea but after changing to
if (!self.Trait<Mobile>().CanEnterCell(pos)),
the dead lock is back. I think it is because the harvester thinks it is occupied by ally and it thinks it can mine from the cell.